src/utils/contribution.js

```diff
@@ -60,51 +85,228 @@ export default class Contribution {
+   * @param  {String} field field to check
    */
-  static getValue(component, changes, field) {
-    return changes && changes[field] ? changes[field] : this.getOriginalValue(component, field) || ''
+  static getValueAndIfDifferent(component, preview, field) {
+    const originalValues = get(component, field)
+    if (preview && get(preview, field)) {
+      return {
+        value: get(preview, field),
+        isDifferent: true
+      }
+    }
+    if (!originalValues || !originalValues.length) {
+      return [
+        {
+          value: 'core',
+          isDifferent: false
+        }
+      ]
+    }
+    return originalValues.map(value => ({
+      value: value || 'core',
+      isDifferent: false
+    }))
   }
+
   /**
    * Verify any difference between changes values and original values
    * If true, return the true statement
    * @param  {} component original component
-   * @param  {} changes object containing changes
+   * @param  {} preview object containing changes
    * @param  {} field field to check
    * @param  {} then_ condition returned if true
    * @param  {} else_ condition returned if false
    */
-  static ifDifferent(component, changes, field, then_, else_) {
-    return changes && changes[field] && !isEqual(changes[field], this.getOriginalValue(component, field))
+  static ifDifferent(component, preview, field, then_, else_) {
+    return preview &&
+      this.getUpdatedValue(preview, field) &&
+      !isEqual(this.getUpdatedValue(preview, field), this.getOriginalValue(component, field))
       ? then_
       : else_
   }
+
   /**
    * Return a specific class name if the condition of difference it true
    * @param  {} component original component
-   * @param  {} changes object containing changes
+   * @param  {} preview object containing changes
    * @param  {} field field to check
    * @param  {} className className to apply if the field has some changes
    */
-  static classIfDifferent(component, changes, field, className) {
-    return this.ifDifferent(component, changes, field, className, '')
+  static classIfDifferent(component, preview, field, className) {
+    return this.ifDifferent(component, preview, field, className, '')
   }
+
+  /**
+   * Check if the preview has some changes from the original definitions
+   * Returns an object containing each change
+   * @param {*} definition
+   * @param {*} preview
+   * @return {Object} Return a new object which represents the diff
+   */
+  static getChangesFromPreview(definition, preview) {
+    if (isEmpty(definition) || isEmpty(preview)) return
+    return this.difference(preview, definition)
+  }
+
+  /**
+   * Deep diff between two objects, using lodash
+   * @param  {Object} object Object compared
+   * @param  {Object} base   Object to compare with
+   * @return {Object} Return a new object which represents the diff
+   */
+  static difference(object, base) {
```

Comment:
See if `deep-diff` can help here. we use it already in `CurationReview`. That code is largely dead given the new work you all are doing but the diffing is still relevant.

Reply:
Thank you @jeffmcaffer , i needed to update the library since the recent version of deep-diff is 1.0.2 (the 0.3.8 that was in the package.json has some incompatibilities).
This saves me a lot of work / code.

Reply:
Awesome. Great to update the versions as we go as well.